hermes-common/src/main/java/pl/allegro/tech/hermes/infrastructure/zookeeper/cache/CacheListeners.java

```diff
@@ -0,0 +1,33 @@
+package pl.allegro.tech.hermes.infrastructure.zookeeper.cache;
+
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+class CacheListeners {
+
+    private static final Logger logger = LoggerFactory.getLogger(CacheListeners.class);
+
+    private final List<Consumer<PathChildrenCacheEvent>> callbacks = new ArrayList<>();
+
+    // NOTE: we probably don't need to be bothered by this synchronization, as adding occurs only on startup
+    // and this is called relatively rare
+    synchronized void addListener(Consumer<PathChildrenCacheEvent> callback) {
```

Comment:
I think we can put callbacks into `ConcurrentLinkedQueue` and don't care how and when listeners are added.